### ISSUE TITLE
[a11y]: OverflowMenu [default] - Expand/Collapse status is not announced

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15258]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.40.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.39.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
Once the user interacting , 
Expand/Collapse  status is not announced by screen reader 

Screen reader expected to announce both Expand & Collapse status, make it   possible  by using ARIA Expand /Collapse properties accordingly.

### Recommended Fix
When the name, role or state of an element is not conveyed appropriately, it prevents screen reader users to interact smoothly with the page element. 

### WCAG 2.1 Violation
4.1.2 Name, Role, Value

**Comment by guidari** [comment]
This is the same scneario as `Search` component in this [issue](https://github.com/carbon-design-system/carbon/issues/14817)

The `aria-expanded` attribute is in the `IconButton` component and when we click to expand and show the Menu, the focus is redirected to the Menu, skipping the `aria-expanded` to be read. 
Since we want to keep this focus functionality the `expanded` announce won't be possible.

But when we open the `OverflowMenu` it announces the menu-item, so the user knows that the Menu is open.
